src/toil/wdl/toilwdl.py

```diff
@@ -0,0 +1,2098 @@
+        '''
+        For all Array[File] inputs, there are a couple of recipes to import each
+        of the files inside into the toil jobStore properly, and this function
+        handles writing those.
+
+        :param sep: Whether an array should be concatenated into a large string
+                    with some separator.  Equivalent to ' '.join(filearray).
+        :param sep_var: The string separator used to join the array of strings.
+                        i.e. ' -V ' for something like ' -V '.join(filearray)
+        :param docker: If this is a docker function, in which case the filepaths
+                       need to begin with the default of '/data'.
+        :param job_declaration_key:
+        :param cmd_name:
+        :return: A string writing all of this.
+        '''
+        fn_section = ''
+
+        if job_declaration_key:
+            formatted_key = '["' + job_declaration_key + '"]'
+        else:
+            formatted_key = ''
+
+        if called_multiple_times:
+            formatted_key = ''
+            multicall_key = '["' + job_declaration_key + '"]'
+        else:
+            multicall_key = ''
+
+        if docker:
+            path_appended = '"/data/" + i[1]'
+        else:
+            path_appended = 'j'
+
+        if sep:
+            fn_section = fn_section + '    ' + cmd_name + '_list = []\n'
+        fn_section = fn_section + '    for i in ' + cmd_name + formatted_key + ':\n'
+        fn_section = fn_section + '        try:\n'
+        fn_section = fn_section + '            j = job.fileStore.readGlobalFile(i' + multicall_key + '[0], userPath=os.path.join(tempDir, i' + multicall_key + '[1]))\n'
+        if sep:
+            fn_section = fn_section + '            ' + cmd_name + '_list.append(' + path_appended + ')\n'
+        fn_section = fn_section + '        except:\n'
+        fn_section = fn_section + '            j = os.path.join(tempDir, i' + multicall_key + '[1])\n'
+        if sep:
+            fn_section = fn_section + '            ' + cmd_name + '_list.append(' + path_appended + ')\n'
+            fn_section = fn_section + '    ' + cmd_name + '_sep = "' + str(sep_var) + '".join(' + cmd_name + '_list)\n\n'
+        return fn_section
+
+    def write_function_cmdline(self, job, docker):
+        '''
+        Write a series of commandline variables to be concatenated together
+        eventually and either called with subprocess.Popen() or with
+        apiDockerCall() if a docker image is called for.
+
+        :param job: A list such that:
+                        (job priority #, job ID #, Job Skeleton Name, Job Alias)
+        :return: A string representing this.
+        '''
+
+        fn_section = ''
+        command_var_decl_array = []
+        job_task_reference = job[2]
+        for cmd in self.tasks_dictionary[job_task_reference]['raw_commandline']:
+            cmd_name = cmd[0]
+            cmd_type = cmd[1]
+            cmd_actions_dict = cmd[2]
+     
```

Comment:
May want to document here (or somewhere else) that this won't work with `${default="foo" bar}` or `${true="foo" false="bar" Boolean baz}`.